integration_tests/src/main/python/cast_test.py

```diff
@@ -153,32 +178,65 @@ def test_cast_string_date_non_ansi():
                         ids=idfn)
 @tz_sensitive_test
 @allow_non_gpu(*non_utc_allow)
-def test_cast_string_ts_valid_format(data_gen):
+def test_cast_string_ts_valid_format_ansi_off(data_gen):
     # In Spark 3.2.0+ the valid format changed, and we cannot support all of the format.
     # This provides values that are valid in all of those formats.
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).select(f.col('a').cast(TimestampType())),
-            conf = {'spark.rapids.sql.hasExtendedYearValues': 'false',
-                'spark.rapids.sql.castStringToTimestamp.enabled': 'true'})
+            conf = copy_and_update(ansi_disabled_conf,
+                                   {'spark.rapids.sql.hasExtendedYearValues': False,
+                                    'spark.rapids.sql.castStringToTimestamp.enabled': True}))
+
+
+@pytest.mark.skip(reason="https://github.com/NVIDIA/spark-rapids/issues/11556")
+@pytest.mark.parametrize('data_gen', [StringGen(date_start_1_1_1)],
+                         ids=idfn)
+@tz_sensitive_test
+@allow_non_gpu(*non_utc_allow)
+def test_cast_string_ts_valid_format_ansi_on(data_gen):
+    # In Spark 3.2.0+ the valid format changed, and we cannot support all of the format.
+    # This provides values that are valid in all of those formats.
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : unary_op_df(spark, data_gen).select(f.col('a').cast(TimestampType())),
+        conf = copy_and_update(ansi_enabled_conf,
+                               {'spark.rapids.sql.hasExtendedYearValues': False,
+                                'spark.rapids.sql.castStringToTimestamp.enabled': True}))
+
 
 @allow_non_gpu('ProjectExec', 'Cast', 'Alias')
 @pytest.mark.skipif(is_before_spark_320(), reason="Only in Spark 3.2.0+ do we have issues with extended years")
-def test_cast_string_date_fallback():
+def test_cast_string_date_fallback_ansi_off():
+    """
+    This tests that STRING->DATE conversion is run on CPU, via a fallback.
+    The point of this test is to exercise the fallback, and not to examine any errors in casting.
+    There is no change in behaviour between Apache Spark and the plugin, since they're both
```

Comment:
nitpicking but what if the fallback does not happen in the ANSI-enabled mode?

Reply:
Once the plugin adds support for string->date casting, we will have to modify this test.  At a minimum, there won't be a fallback.  I don't foresee a case where we might support ANSI=on and *not* ANSI=off, but that's hard to predict.

I'd postpone any further futzing with this until the plugin's behaviour changes.